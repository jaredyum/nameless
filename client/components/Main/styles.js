export default theme => ({
  main: {
    flex: 1,
    display: 'flex',
    flexDirection: 'column',
    maxWidth: '100%',
    [theme.breakpoints.up('md')]: {
      maxWidth: 'calc(100% - 240px)'
    }
  },
  toolbar: theme.mixins.toolbar
});
