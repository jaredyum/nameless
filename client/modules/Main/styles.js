export default theme => ({
  main: {
    flex: 1,
    display: 'flex',
    flexDirection: 'column'
  },
  toolbar: theme.mixins.toolbar
});
