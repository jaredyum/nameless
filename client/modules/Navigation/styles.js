/**
 * The default width of the drawer (sidebar).
 * @const {number}
 */
const DRAWER_WIDTH = 240;

/**
 * The parent styles for this modules.
 * @param {!Object} theme The MUI theme object.
 * @return {!Object.<string, !Object>} The generated styles.
 */
export default theme => ({
  active: {
    fontWeight: 'bold'
  },
  appBar: {
    zIndex: theme.zIndex.drawer + 1
  },
  drawerPaper: {
    width: DRAWER_WIDTH,
    [theme.breakpoints.up('md')]: {
      position: 'relative'
    }
  },
  flex: {
    flex: 1
  },
  link: {
    color: '#333',
    cursor: 'pointer',
    width: '100%',
    textDecoration: 'none',
    padding: '5px 0'
  },
  menuItem: {
    height: 'auto',
    padding: 0,
    color: '#333',
    cursor: 'pointer',
    width: '100%',
    textDecoration: 'none'
  },
  navIconHide: {
    [theme.breakpoints.up('md')]: {
      display: 'none'
    }
  },
  searchBox: {
    borderRadius: '5px',
    padding: '4px 0 2px 10px',
    border: '1px solid #ccc',
    textAlign: 'left'
  },
  toolbar: {
    backgroundColor: theme.palette.primary.main,
    color: '#fff',
    ...theme.mixins.toolbar
  }
});
