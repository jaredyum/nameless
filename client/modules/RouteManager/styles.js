export default {
  routeManager: {
    flex: 1,
    display: 'flex'
  }
};
