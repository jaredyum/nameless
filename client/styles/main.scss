// Vendors
@import './vendors/normalize';
@import './vendors/reset';

// Modules
@import './modules/defaults';
