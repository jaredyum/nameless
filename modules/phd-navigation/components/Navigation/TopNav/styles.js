export default theme => ({
  appBar: {
    zIndex: theme.zIndex.drawer + 1
  },
  navIconHide: {
    [theme.breakpoints.up('md')]: {
      display: 'none'
    }
  },
  flex: {
    flex: 1
  },
  searchBox: {
    borderRadius: '5px',
    padding: '4px 0 2px 10px',
    border: '1px solid #ccc',
    textAlign: 'left'
  }
});
