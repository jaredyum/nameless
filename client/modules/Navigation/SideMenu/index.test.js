// MUI Components
import Drawer from '@material-ui/core/Drawer';
import Toolbar from '@material-ui/core/Toolbar';

// Utils
import {
  getComponentWrapper,
  testCommonComponentAttrs
} from 'utils/tests';

// Components
import SideMenuGroups from './partials/side-menu-groups';
import { SideMenuJest } from './index';

// Local styles
import styles from '../styles';

const defaultProps = {
  classes: styles
};

let wrapper;
let drawers;

describe('<SideMenu /> Component', () => {
  testCommonComponentAttrs(SideMenuJest, defaultProps);

  beforeEach(() => {
    wrapper = getComponentWrapper(SideMenuJest, defaultProps);
    drawers = wrapper.find(Drawer);
  });

  describe('basic rendering', () => {
    it('should render 2 drawers, one for desktop, one for mobile', () => {
      expect(drawers).toHaveLength(2);
    });

    it('should have a Toolbar and SideMenuGroups', () => {
      const drawer = drawers.first();
      expect(drawer.find(Toolbar)).toHaveLength(1);
      expect(drawer.find(SideMenuGroups)).toHaveLength(1);
    });
  });
});
