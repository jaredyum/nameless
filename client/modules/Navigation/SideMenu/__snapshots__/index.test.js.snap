// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideMenu /> Component when rendering should render a component matching the snapshot 1`] = `
<div>
  <Hidden
    implementation="js"
    lgDown={false}
    lgUp={false}
    mdDown={false}
    mdUp={true}
    smDown={false}
    smUp={false}
    xlDown={false}
    xlUp={false}
    xsDown={false}
    xsUp={false}
  >
    <WithStyles(Drawer)
      ModalProps={
        Object {
          "keepMounted": true,
        }
      }
      classes={
        Object {
          "paper": undefined,
        }
      }
      variant="temporary"
    >
      <div>
        <WithStyles(Toolbar)>
          <WithStyles(Typography)
            color="inherit"
            variant="title"
          />
        </WithStyles(Toolbar)>
        <WithStyles(Divider) />
        <SideMenuGroups />
      </div>
    </WithStyles(Drawer)>
  </Hidden>
  <Hidden
    implementation="css"
    lgDown={false}
    lgUp={false}
    mdDown={false}
    mdUp={false}
    smDown={true}
    smUp={false}
    xlDown={false}
    xlUp={false}
    xsDown={false}
    xsUp={false}
  >
    <WithStyles(Drawer)
      classes={
        Object {
          "paper": undefined,
        }
      }
      open={true}
      variant="permanent"
    >
      <div>
        <WithStyles(Toolbar)>
          <WithStyles(Typography)
            color="inherit"
            variant="title"
          />
        </WithStyles(Toolbar)>
        <WithStyles(Divider) />
        <SideMenuGroups />
      </div>
    </WithStyles(Drawer)>
  </Hidden>
</div>
`;
